omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java

```diff
@@ -464,6 +464,18 @@ public static void setValue(Obs obs, Object value) throws ParseException, Conver
 						}
 					}
 				}
+				else if (obs.getConcept().getDatatype().containsDate() || obs.getConcept().getDatatype().containsTime()) {
```

Comment:
Take a second look at the ticket description.

Reply:
@dkayiwa , it seems there some thing you want me to see ,but am not seeing ..can you be a little more specific on what exactly you want me to adjust

Reply:
Obs.setValueDateTime(Date) or Obs.setValueDate() or Obs.setValueTime() are basically the same methods , theyjust do the same thing

Reply:
@mozzy11, why would you allow creating an observation that has no date part?

Reply:
Hi @mozzy11, is there any more thing that's probably not yet clear here?

Reply:
@Ruhanga what exactly do you mean by an obs with no date part ??
This check here
`if (obs.getConcept().getDatatype().containsDate() || obs.getConcept().getDatatype().containsTime())` only allows creation of obs with a date part or a time part

Reply:
Hi @mozzy11, so when it's a time posted, the date part is lost I believe? The ticket talks of multiple time zone for the different date formats, that's if the time is part of the date. Not time formats **or** ( **||** ) date formats but rather both time and date or simply date (By date I mean day, month, year) for different time zones and formats (date and time). That's my understanding. Does this make sense?

Reply:
ok , kind of makes sense. . let me try that out . thanks. i actually hadn't considered that little simple detail

Reply:
@Ruhanga , check out the fix i have done. check the ticket comments for a detailed explanation of what i did